File: lab_2/IS_Lab2_Python/json_serializer.py
import json

from data_extractor import DataExtractor


class JsonSerializer:

    @staticmethod
    def run(deserialized_data, file_location):
        extracted_data = DataExtractor.extract(deserialized_data)

        json_temp = {"departaments": extracted_data}
        with open(file_location, 'w', encoding='utf-8') as f:
            json.dump(json_temp, f, ensure_ascii=False)

File: lab_2/IS_Lab2_CSharp/IS_Lab2_CSharp/JsonSerializer.cs
using Newtonsoft.Json;

namespace IS_Lab2_Extra;

public static class JsonSerializer
{
    public static void Run(List<Dictionary<string, object?>> deserializedData, string filePath)
    {
        var extractedData = DataExtractor.Extract(deserializedData);
        var serializedList = JsonConvert.SerializeObject(new {departaments = extractedData});
        var path = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, @"..\..\..\", filePath));
        File.WriteAllText(path, serializedList);
    }
}